apps/dapp/src/modules/auction/hooks/use-has-private-key.ts

```diff
@@ -0,0 +1,33 @@
+import { abis } from "@axis-finance/abis";
+import { Address } from "viem";
+import { useReadContract } from "wagmi";
+
+type UseHasPrivateKeyParameters = {
+  address: Address;
+  lotId: string;
+  chainId: number;
+  enabled?: boolean;
+};
+
+export function useHasPrivateKey({
+  lotId,
+  address,
+  chainId,
+  enabled,
+}: UseHasPrivateKeyParameters) {
+  const { data, ...query } = useReadContract({
+    address,
+    chainId,
+    abi: abis.encryptedMarginalPrice,
+    functionName: "auctionData",
+    args: [BigInt(lotId)],
+    query: { enabled },
+  });
+
+  const privateKey = data?.[9] !== 0n;
```

Comment:
Can we add a comment here that states that when a private key is not yet submitted the smart contract stores 0n in this data field?